exercises/dominoes/src/test/kotlin/DominoesTest.kt

```diff
@@ -0,0 +1,158 @@
+import org.junit.Rule
+import org.junit.rules.ExpectedException
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class DominoesTest {
+
+    @Rule
+    @JvmField
+    var expectedException = ExpectedException.none()
+
+    @Test
+    fun `empty input = empty output`() {
+        val dominoesList = emptyList<Domino>()
+        val chain = Dominoes.formChain(dominoesList)
+        assertEquals(0, chain.size, "The output list should be empty.")
+    }
+
+    @Test
+    fun `singleton input = singleton output`() {
+        val dominoesList = listOf(Domino(1, 1))
+        val chain = Dominoes.formChain(dominoesList)
+        assertValidChain(dominoesList, chain)
+    }
+
+    @Test
+    fun `singleton can't be chained`() {
+        val dominoesList = listOf(Domino(1, 2))
+        expectedException.expect(ChainNotFoundException::class.java)
+        expectedException.expectMessage("No domino chain found.")
+        Dominoes.formChain(dominoesList)
+    }
+
+    @Test
+    fun `three elements`() {
+        val dominoesList = listOf(Domino(1, 2), Domino(3, 1), Domino(2, 3))
+        val chain = Dominoes.formChain(dominoesList)
+        assertValidChain(dominoesList, chain)
+    }
+
+    @Test
+    fun `can reverse dominoes`() {
+        val dominoesList = listOf(Domino(1, 2), Domino(1, 3), Domino(2, 3))
+        val chain = Dominoes.formChain(dominoesList)
+        assertValidChain(dominoesList, chain)
+    }
+
+    @Test
```

Comment:
Please, Use
```kotlin
@Test (expected=ChainNotFoundException::class)
```
and avoid `var expectedException`. We do not force a student to use a precise error message. The fact that exception is thrown is sufficient. 

I'd also recommend to have a look at updated tests, for example at *robot-simulator/RobotTest*. I can give you an idea of pleasure of local dsl and single-assert tests. Maybe you'll find it interesting for further test set development. It's not mandatory, though.

Reply:
Nice practice for me. Please have a look at the updated test.